context/src/test/java/io/opentelemetry/context/LazyStorageTest.java

```diff
@@ -13,37 +13,36 @@
 import java.io.Writer;
 import java.net.URL;
 import java.util.concurrent.atomic.AtomicReference;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
+import org.junitpioneer.jupiter.ClearSystemProperty;
+import org.junitpioneer.jupiter.SetSystemProperty;
 
 class LazyStorageTest {
 
   private static final String CONTEXT_STORAGE_PROVIDER_PROPERTY =
       "io.opentelemetry.context.contextStorageProvider";
+  private static final String MOCK_CONTEXT_STORAGE_PROVIDER =
+      "io.opentelemetry.context.LazyStorageTest$MockContextStorageProvider";
   private static final AtomicReference<Throwable> DEFERRED_STORAGE_FAILURE =
       new AtomicReference<>();
 
-  @AfterEach
-  public void after() {
-    System.clearProperty(CONTEXT_STORAGE_PROVIDER_PROPERTY);
-  }
-
   @Test
-  public void empty_provides() {
+  @ClearSystemProperty(key = CONTEXT_STORAGE_PROVIDER_PROPERTY)
+  public void empty_providers() {
```

Comment:
Just one last nit, if you can change the methods to be package-private, since junit5, that would be swell, not a required change though.

Reply:
Thanks. That was done.